cedar-policy-core/src/ast/value.rs

```diff
@@ -299,6 +287,44 @@ pub struct Set {
 }
 
 impl Set {
+    /// Create an empty set
+    pub fn empty() -> Self {
+        Self {
+            authoritative: Arc::new(BTreeSet::new()),
+            fast: Some(Arc::new(HashSet::new())),
+        }
+    }
+
+    /// Create a set with the given `Value`s as elements
+    pub fn new(vals: impl IntoIterator<Item = Value>) -> Self {
+        let authoritative: BTreeSet<Value> = vals.into_iter().collect();
+        let fast: Option<Arc<HashSet<Literal>>> = authoritative
+            .iter()
+            .map(|v| v.try_as_lit().cloned())
+            .collect::<Option<HashSet<Literal>>>()
+            .map(Arc::new);
+        Self {
+            authoritative: Arc::new(authoritative),
+            fast,
+        }
+    }
+
+    /// Create a set with the given `Literal`s as elements
+    pub fn of_lits(lits: impl IntoIterator<Item = Literal>) -> Self {
```

Comment:
[nit] Isn't `from_lits` more idiomatic?

Reply:
You are correct.  I think I got sidetracked by wanting to write `Set::of_lits` which reads well in English

Reply:
Updated